client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -94,6 +98,7 @@
         this.selector = new SegmentSelector(stream, controller, outputStreamFactory, config);
         this.serializer = serializer;
         this.config = config;
+        this.backgroundThreadPool = backgroundThreadPool;
```

Comment:
I think we should add `Preconditions.checkNotNull` for the `config` and the two executors params.

Reply:
Done